mypy error in `op_invocation.py` because of incorrect type hint

### Dagster version

1.6.11

### What's the issue?

When we run `mypy` on a file importing dagster, we get the `Invalid "type: ignore" comment` error.

`python --version` --> 3.9.18
`mypy --version` --> 0.991

### What did you expect to happen?

The `mypy --strict` command should pass without any errors. 
As a workaround, we've added `ignore_errors = True` in mypy config file for `dagster._core.definitions.*` package.

### How to reproduce?

> test_file.py
```python
import dagster
print(dagster.__version__)
```

> mypy command
```bash
mypy --strict test_file.py
```

> errors
```
site-packages/dagster/_core/definitions/op_invocation.py:225: error: Invalid "type: ignore" comment  [syntax]
site-packages/dagster/_core/definitions/op_invocation.py:229: error: Invalid "type: ignore" comment  [syntax]
site-packages/dagster/_core/definitions/op_invocation.py:237: error: Invalid "type: ignore" comment  [syntax]
site-packages/dagster/_core/definitions/op_invocation.py:239: error: Invalid "type: ignore" comment  [syntax]
Found 4 errors in 1 file (checked 1 source file)
```

### Deployment type

Local

### Deployment details

_No response_

### Additional information

The bug is likely introduced in this [PR](https://github.com/dagster-io/dagster/pull/19482/files). This PR adds a `# type: ignore (pyright bug)` statement where the additional comment along with type ignore annotation is not supported by all type checkers.

According to the PEP-484 documentation, it is mentioned [here](https://peps.python.org/pep-0484/#:~:text=In%20some%20cases,or%20other%20marker%3E) that the comments in `# type: ignore` must be preceded by a `#`. The correct fix would be replacing the type-hint with `# type: ignore # (pyright bug)`.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

--- a/python_modules/dagster/dagster/_core/definitions/op_invocation.py
+++ b/python_modules/dagster/dagster/_core/definitions/op_invocation.py
@@ -222,11 +222,11 @@ def direct_invocation_result(
         # try-except handles "vanilla" asset and op invocation (generators and async handled in
         # _type_check_output_wrapper)
 
-        input_dict = _resolve_inputs(op_def, input_args, input_kwargs, bound_context)  # type: ignore (pyright bug)
+        input_dict = _resolve_inputs(op_def, input_args, input_kwargs, bound_context)  # type: ignore # (pyright bug)
 
         result = invoke_compute_fn(
             fn=compute_fn.decorated_fn,
-            context=bound_context,  # type: ignore (pyright bug)
+            context=bound_context,  # type: ignore # (pyright bug)
             kwargs=input_dict,
             context_arg_provided=compute_fn.has_context_arg(),
             config_arg_cls=(
@@ -234,9 +234,9 @@ def direct_invocation_result(
             ),
             resource_args=resource_arg_mapping,
         )
-        return _type_check_output_wrapper(op_def, result, bound_context)  # type: ignore (pyright bug)
+        return _type_check_output_wrapper(op_def, result, bound_context)  # type: ignore # (pyright bug)
     except Exception:
-        bound_context.unbind()  # type: ignore (pyright bug)
+        bound_context.unbind()  # type: ignore # (pyright bug)
         raise
 
 

